openhcl/underhill_core/src/worker.rs

```diff
@@ -2588,14 +2588,35 @@ async fn new_underhill_vm(
             .unwrap_or(!controllers.mana.is_empty());
         tracing::info!(enable_mnf, "Underhill MNF enabled?");
 
+        let max_version = env_cfg
+            .vmbus_max_version
+            .map(vmbus_core::MaxVersionInfo::new)
+            .or_else(|| {
+                // For compatibility with rollback, the client ID feature is currently disabled,
+                // except for isolated guests which do not support servicing.
+                (!hardware_isolated).then_some(vmbus_core::MaxVersionInfo {
+                    version: u32::MAX,
+                    feature_flags: vmbus_core::protocol::FeatureFlags::all().with_client_id(false),
```

Comment:
I'd suggest we be explicit about both the max version and the feature flags, rather than just strip the client ID bit. That way, we don't have another regression later if we fail to put a more general fix in for this.